### PR TITLE
modes/ctr128.c: fix false carry in counter increment procedure.

### PR DESCRIPTION
**Pull request by dot-asm**
GH issue #1916 affects only big-endian platforms. TLS is not affected, because TLS fragment is never big enough.

Target is all supported releases, i.e. down to 1.0.1. First commit, a test case, goes only down to 1.1.0, because it probably won't apply to 1.0.x. In case second commit can't be cherry-picked, there will be separate request for 1.0.x.

### CODE REVIEW

File: test/evptests.txt [hunk 1]
```diff
@@ -791,6 +791,13 @@ Plaintext = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F
 Ciphertext = 5104A106168A72D9790D41EE8EDAD388EB2E1EFC46DA57C8FCE630DF9141BE28
 
 Cipher = aes-128-ctr
+Key = 7E24067817FAE0D743D6CE1F32539163
+IV = 00000000000000007FFFFFFFFFFFFFFF
+Operation = ENCRYPT
+Plaintext = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F
+Ciphertext = A2D459477E6432BD74184B1B5370D2243CDC202BC43583B2A55D288CDBBD1E03
+
+Cipher = aes-128-ctr
```
Comment (mattcaswell) [inline test/evptests.txt]: Where did you get this test vector from? It is in the section labelled "AES Counter test vectors from RFC3686", but I can't see this one in that RFC.

Comment (dot-asm) [inline test/evptests.txt]: I copied one vector from above, made up IV that would suffer from problem in question, and collected ciphertext output from platform/implementation that doesn't suffer from problem in question.

Comment (dot-asm) [inline test/evptests.txt]: "platform/implementation" means that there are fault-free implementations on big-endian platforms, most notably those that rely on dedicated AES instructions.

Comment (mattcaswell) [inline test/evptests.txt]: In that case I think a comment in front of this new vector would be helpful to say that this is an "extra" vector not in RFC3686 generated by us on a "known good" platform to test carrying in IVs

**Pull request closed without merging**